var/spack/repos/builtin/packages/quinoa/package.py

```diff
@@ -0,0 +1,65 @@
+##############################################################################
+# Copyright (c) 2017, Los Alamos National Security, LLC
+# Produced at the Los Alamos National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Quinoa(CMakePackage):
+    """Quinoa is a set of computational tools that enables research and
+       numerical analysis in fluid dynamics. At this time it is a test-bed
+       to experiment with various algorithms using fully asynchronous runtime
+       systems.
+    """
+
+    homepage = "http://quinoacomputing.org"
+    url      = "https://github.com/quinoacomputing/quinoa/tarball/Quinoa_v0.1"
+
+    root_cmakelists_dir = '../src'
+
+    version('develop', git='https://github.com/quinoacomputing/quinoa', branch='master')
+
+    variant('debug', default=False, description='Build debug version')
+
+    depends_on('hdf5+mpi')
+    depends_on("charm+mpi+shared~netlrts")
```

Comment:
A package that fails with its default Spack variants is a problem.  Looking at `charm`, I think that the problems should be fixed in that package.

If we had #2386, I would suggest that `+mpi` and `+netlrts` get merged into a single multi-valued variant called `backend`, with values `mpi` or `netlrts` or `verbs`.  Then everything would work pretty smoothly.

Without #2386... can you disable all backends by default in `charm/package.py`?  Can you please also default `shared=True`?  With these changes, you can depend on `charm+mpi`.

But that brings up the question... will `quinoa` work when Charm is built with other backends?  If so, it should just depend on `charm`.  It should not try to forward a backend variant to `charm`, the user should set that directly on `charm` if needed.  If you prefer `charm+mpi`, you can set that in your `packages.yaml` file.  (We're working on other ways to do this too).